UKHomeOffice/nmsw-ui
# NMSW-126 Create Cookie Preference form

# Ticket

NMSW-126

----

## AC

Given I am viewing the cookie page
When I view the manage section
Then I am provided a form to manage my preferences
AND it should show my current preference

Given I have not selected a cookie preference
When I view the manage my preference form
Then the form should be set to no

Given I am viewing the manage preferences form
When I have previously selected Yes to analytics cookies
Then the form should be set to Yes

Given I am viewing the manage preferences form
When I have previously selected No to analytics cookies
Then the form should be set to No

Give I am viewing the manage preferences form
When I change the selected preference
And click save cookie settings
Then my new preference for analytics cookies should be saved

----

## To test

- To Test
- run app locally
- go to localhost:3000
- delete the cookiePreference cookie if it exists
- click the footer link to the cookie page
- > you should see a manage my cookie preferences forms
- the form should be set to no
- go to the dashboard
- go to the cookie banner at the top of the page
- click the accept button
- click the change your cookie settings link
- > you should be taken to the cookie page
- the form should be set to yes
- go to the dev tools and view the cookies section under application
- > the cookiePreference cookie should be set to yes
- change the answer to no
- click save
- the form should be set to no
- go to the dev tools and view the cookies section under application
- > the cookiePreference cookie should be set to no
- delete the cookiePreference cookie and refresh the page
- go to the cookie page
- change the form to yes
- click save
- > the cookie banner at the top of the screen should disappear
- delete the cookiePreference cookie again and refresh the page 
- go to the cookie page
- go to the cookie banner at the top of the page
- click accept
- on the cookie confirmation click hide cookie banner
- > the page should reload and the cookie form should be set to yes
----

## Notes

You should not see a success banner (this work will be completed in NMSW-126
